src/main/java/wolox/training/controller/UserController.java

```diff
@@ -0,0 +1,158 @@
+package wolox.training.controller;
+
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.UserException;
+import wolox.training.exceptions.UserIdMismatchException;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.model.User;
+import wolox.training.repository.BookRepository;
+import wolox.training.repository.UserRepository;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    private final UserRepository userRepository;
+
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public UserController(UserRepository userRepository, BookRepository bookRepository) {
+        this.userRepository = userRepository;
+        this.bookRepository = bookRepository;
+    }
+
+    /**
+     * This method gets list of {@link User}
+     */
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    public List<User> findAll() {
+        return userRepository.findAll();
+    }
+
+    /**
+     * This method gets one {@link User} by username
+     *
+     * @param username Username of the user (String)
+     *
+     * @return got {@link User} for username.
+     * @throws UserNotFoundException if there is no user associated with that username
+     */
+    @GetMapping("/username")
```

Comment:
This endpoint will fetch only an user? I think this get should be by id if it returns only an user (as a resource request), or it should return a list of matches if it is a search by name or some other non unique field